0006-cdt-protocol/README.md

```diff
@@ -0,0 +1,205 @@
+---
+name: Configuration Data Transport protocol
+shortname: 6/CDT
+status: draft
+editor: Andrew Pasika <apasika@cybervisiontech.com>
+---
+
+- [Introduction](#introduction)
+- [Language](#language)
+- [Requirements and constraints](#requirements-and-constraints)
+- [Design](#Design)
+  - [Configuration pull](#configuration-pull)
+    - [Subject structure](#subject-structure)
+    - [Targeted message types](#targeted-message-types)
+  - [Configuration push](#configuration-push)
+    - [Subject structure](#subject-structure)
+
+## Introduction
+
+Configuration Data Transport (CDT) protocol is designed to communicate endpoint configuration data between Kaa services.
+
+In Kaa architecture, some services are designed to manage endpoint configuration in different ways — store it, process it, publish it, etc.
+One of the core features in such context is the ability to send and receive endpoint configurations.
+
+For this purpose, the CDT protocol is designed.
+It defines the endpoint configuration data structure so that it's interpreted by all involved services in the same way.
+
+In CDT lingo, sending an endpoint configuration to a service is called *configuration push*, while receiving it from a service is called *configuration pull*.
+
+To effectively perform these activities, endpoint configuration data should be communicated in a format recognized by all the services involved.
+To achieve this, CDT protocol was designed based on the [3/Messaging IPC][3/MIPC] protocol.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+The following terms and definitions are used in this RFC.
+
+- **Configuration push**: sending an endpoint configuration data to a service.
+- **Configuration pull**: receiving an endpoint configuration data from a service.
+- **EP configuration provider (provider)**: any service that sends an endpoint configuration to the other service.
+- **EP configuration consumer (consumer)**: any service that receives an endpoint configuration from the other service.
+
+## Requirements and constraints
+
+CDT requirements:
+
+- Once a new endpoint configuration is pushed, there must be an EP configuration provider broadcasting an event over [NATS](http://nats.io/) about the new configuration availability.
+- The provider must receive a confirmation that the EP configuration was successfully pushed to the consumer.
```

Comment:
Rephrase to: "In case of a configuration push, the provider must receive a receipt acknowledgement from the consumer"